### DISCUSSION TITLE
LedConfiguration on (XY-WFUSB)

### DISCUSSION THREAD

**Discussion by doofmars** [arendst/Tasmota#17954] [Q&A] [answered]
Hello,

In the [Components](https://tasmota.github.io/docs/Components/) documentation there are several Led components (288-323) are listed.

1. What does the "4xLeds" Comment next to any Led components mean?
2. This seems to relate to the select option next to the GPIO Configuration, if Led is selected another value can be selected (1-4) but It is unclear to me what changing this value does.
3. Further questions would go too much into the hardware side of my device and do not require an answer.

### How I stumbled on this:

I just flashed my device (XY-WFUSB aka. Sinilink USB Switch Module) with Tasmota 12.3.1 and want to configure the LED.
The device has 3 LEDs to show the current operation mode. The following is a list with my observations:

1. Blue used during flashing and boot (is off after the device is started).
5. Green used if the power is "off"
6. Red used if the power is "on"

Details about this device and tasmotize instructions can be found here https://templates.blakadder.com/sinilink_XY-WFUSB.html

The configuration I used from the page above:
```json
{"NAME":"XY-WFUSB","GPIO":[1,1,0,1,32,224,0,0,0,0,320,0,544,0],"FLAG":0,"BASE":18}
```

First I had a look into the configuration, *Template parameters*. Here `GPIO14` is set to `Led_i` and `1`. 
I discovered that `Led_i` means inverted not sure why this was chosen in the configuration I used.
This confused me and lead me to start writing this question. Especially while queuing the led power the results seem to be wrong.

**Comment by barbudor** [comment, marked as answer]
Is there a question somewhere ?
You are just stating that you have configure the device accordingly to the template.
So what ?

**Reply by doofmars** [reply]
Sorry hit send to early, question is now updated and added. Sorry about the confusion.

**Reply by barbudor** [reply]
I still don't really see a question

A GPIO is configured as "Led_i(nverted)" when the led is ON with a LOW signal (and OFF with a HI signal) which is the case when the led is connected between VCC and the GPIO
When the led is connected between the GPIO and GND, it is ON with a HIGH signal and OFF with a LOW signal and you should use "Led" (non inverted)

The index "1" tell that the led is attached to the Relay "1" output so the LED should be ON when the power is ON

The other Led on GPIO16 is configured to be the link indicator that is flashing when Tasmota tries to connect to Wifi, ON when the WIFI is connected and flashing when Tasmota execute commands.

I have no idea which one of GPIO14 and GPIO16 is the red or the green one

**Reply by doofmars** [reply]
I got confused by the configuration, looked into the documentation and found no answer. And then I asked. This is the answer:

> The index "1" tell that the led is attached to the Relay "1" output so the LED should be ON when the power is ON

How should I know by looking into [the documentation](https://tasmota.github.io/docs/Components/), with the comment 4 x Leds, that this is the case? also that "_i" means inverted? 
The latter I managed to deduce but how the former?

I found another page, [old components](https://tasmota.github.io/docs/Components-old/), that somehow has better descriptions in that regard the comment is `Inverted LED - default state ON`. Not sure why the description is different.

I think the answer to the inverted question is that the red led of this device cannot be controlled by the ESP. The LED is wired to the relay. So I think in order to use the green LED the inverted setting was chosen.

Thanks for the support, even if you did not see my question :D

**Reply by sfromis** [reply]
As the template has both `Led_i` and `LedLink` pins, you can reconfigure the [status leds](https://tasmota.github.io/docs/Lights/#status-leds) and use them however you like. Neither is hardwired to anything.

The text `4 x Leds` is just an indication that there are 4 Led components. Yes, that is a bit redundant when you also have all 4 listed, but the table was generated after extracting data where it was not expanded to Led1, Led2, Led3 and Led4, and then the comment was just taken directly, as most other comments on component usage is a useful short description. The new components table being different is also due to it being generated instead of manually maintained.

**Reply by sfromis** [reply]
I'd even say that the Led_i config in the suggested template is misleading. Neither pin is actually inverted, both will light up when Tasmota turns it to on/high, and off with low. Configuring as inverted or not can be called a matter of taste, about what default state you want. Neither is "wrong" or "right", but maybe a bit unnatural to have one device having different Led behavior from standard Tasmota devices.

Blue is `LedLink`, gpio 16, and green is `Led_i`, gpio 14. Red is indeed not controllable, unless you modify the board.

**Reply by doofmars** [reply]
> The text 4 x Leds is just an indication that there are 4 Led components. 

All I want to say is that, from an outside perspective (me reading the documentation because I don't know what component `Led_i1` is), this indication is not helpful. Not sure how or where to prase this better (especially as it is a generated table). And you could also argue that there are 8 x Relay components ;-)

> Configuring as inverted or not can be called a matter of taste, about what default state you want.

This summarizes my issue quiet good! Thanks.
